### ISSUE TITLE
Support backward hooks in JIT 

### ISSUE THREAD

**Issue by Lilyjjo** [pytorch/pytorch#51969]
## 🚀 Feature
Supporting backward hooks (hooks that are called every time the gradients with respect to module inputs are computed) is a common feature requested from internal users. Eager has fixed their backward hook implementation so now it shouldn't be as difficult to match eager's behavior. 

Eager backwards hooks: https://pytorch.org/docs/master/generated/torch.nn.Module.html#torch.nn.Module.register_full_backward_hook

**Comment by AlphaBetaGamma96** [comment]
Hi, I was wondering if there's been an update on this issue? (Or any other work-around) As I've ran into a use case where I need a JIT-function to call a backward hook but it doesn't get called at all. Thank you! :)

**Comment by Lilyjjo** [comment]
@gmagogsfm routing this to you!

**Comment by AlessandroPolidori** [comment]
@AlphaBetaGamma96 Any news? Is it possible to trace a full_backward_hook in 2023?

**Comment by Tihaw** [comment]
I tried to save the outputs of hidden layer, and then using autograd to calc the hidden layer's gradients.
Then it is good, the save as backward hooks do. But only can readout and calc the gradients, not modifying them.
For my side, JIT is for infer the model rather than training. Inferring stage needs some gradients from the hidden layer, and in this case, backward hooks is not necessary. [https://github.com/pytorch/pytorch/issues/64728](url)

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
